tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/HardwareDeviceLoader.cs

```diff
@@ -92,6 +90,8 @@ public async Task LoadDevices (ILog log, bool includeLocked = false, bool forceR
 						}
 						connectedDevices.Add (d);
 					}
+
+					loaded = true;
```

Comment:
There's a different problem now, if LoadDevices is called multiple times, we'll load devices multiple times:

```
var task1 = loader.LoadDevices (...);
var task2 = loader.LoadDevices (...);
// there are now 2 mlaunch processes
```

Reply:
@rolfbjarne how about the semaphore?

Reply:
Ah, sorry, I was looking at SimulatorLoader

Reply:
I added the Semaphore to HW device loader as well